README.md

```diff
@@ -1,70 +1,135 @@
-* I've included an Xcode project the student can reference as we discuss what it is they need to know in setting up a TableViewController.
-* I would not talk about or include the Navigation Controller in this reading, it will be its own reading after this.
+Believe it or not, these are all the methods that are required to power a table view. The table view can use these methods to get its data set, and will do all the rest of the work itself—drawing the cells, allow the user to scroll through the table view, and so on.
+
+Let's take a look at these methods in more detail.
+
+### Table View Methods
+
+The first method, `numberOfSectionsInTableView(_:)`, returns the number of _sections_ in the tablew view. Many table views only have one section, including the one in this example project, so in this case, the method is simply returning `1`. Table views can be broken into sections, though. You can see this in the Settings app, which groups similar settings categories into sections. Each section is denoted with a large grey area. Sections can have titles, too, although the Settings app doesn't have titles for its sections.
+
+![Settings.app](https://s3.amazonaws.com/learn-verified/ios-settings.png)
+
+In future lessons, you'll deal more with dividing a table view into sections; for the time being, you'll work with table views with just one section.
+
+The next important method is `tableView(_:numberOfRowsInSection:)`. This method gets passed an `Int` representing a particular section of the table view, and returns the number of _rows_ present in that section. Since you're working with table views with only one section in this example, `numberOfRowsInSection` will always be 0 (since sections are 0-indexed, like arrays).
+
+The relevant bit of this method is the single line that returns the number of rows in the section. In this example app, you are creating a list of favorite songs that has been stored in the instance variable `favoriteSongs`. The number of rows—i.e., the number of pieces of data in the data set—is the number of songs stored in the `favoriteSongs` array, so this method simply returns the `count` of that array. Easy, right?
+
+The final method, `tableView(_:cellForRowAtIndexPath:)`, is a bit more complicated. There's a lot going on in this method, so let's break it down bit by bit.
+
+#### Getting a Cell
+
+`tableView(_:cellForRowAtIndexPath:)` does something very important: It returns an actual _cell_ for use in the table view. The "magic" is that, given a cell, the table view will know how to draw it in amongst all the other cells. Neat, huh?
+
+First, `tableView(_:cellForRowAtIndexPath:)` gets passed a `cellForRowAtIndexPath` parameter which lets the method know exactly which cell is being requested.
+
```

Comment:
Me adding onto this, I will merge this in and include this myself.

---

How many times does `tableView(_:cellForRowAtIndexPath:)` get called? 11 times. Lets add a `print()` function inside this function and see? But, I'm going to print the `indexPath` argument that's given to us when the function is called. This `indexPath` constant is of type `NSIndexPath` This `NSIndexPath` object is able to encapsulate very important information. The `NSIndexPath` object has two computed read-only properties which are incredibly helpful for us here. 

`row` - this will give us back the row for which the tableView wants us to give back a cell. If the tableView tell us it wants a cell for the 2nd row, well lets give it back a cell with data for the 2nd row.
`section` - it will give us the section we're in. In our scenario here there will only be one.

Lets print these two computed read-only properties on the `NSIndexPath` object called `indexPath` and see what gets printed to console.

```swift
 override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        
        print("Section: \(indexPath.section) -- Row: \(indexPath.row)")
        
        let cell = tableView.dequeueReusableCellWithIdentifier("basicCell", forIndexPath: indexPath)

        let favoriteSong = favoriteSongs[indexPath.row]
        
        cell.textLabel?.text = favoriteSong

        return cell
    }
```

After running our code, we should see this print to console:
```swift
Section: 0 -- Row: 0
Section: 0 -- Row: 1
Section: 0 -- Row: 2
Section: 0 -- Row: 3
Section: 0 -- Row: 4
Section: 0 -- Row: 5
Section: 0 -- Row: 6
Section: 0 -- Row: 7
Section: 0 -- Row: 8
Section: 0 -- Row: 9
Section: 0 -- Row: 10
```

Well, that `row` property on the `NSIndexPath` object seems like something we could use! Our Array called `favoriteStrings` is zero-based because it's an Array, so whenever this function is called we can take the value of this `row` property and access an element from our `favoriteSongs` array using that value as an index.

The first time this function is called, `row` has a value of `0`.
```swift
let favoriteSong = favoriteSongs[indexPath.row]
// the first time through, the row property is 0. This is the equivalent then of favoriteSongs[0]
// favoriteSong has a value of "Thriller"
```

The second time this function is called, `row` has a value of `1`.
```swift
let favoriteSong = favoriteSongs[indexPath.row]
// the second time through, the row property is 1. This is the equivalent then of favoriteSongs[1]
// favoriteSong has a value of "Never Gonna Give You Up"
```

I hope you can see the pattern now. Arrays and Table Views work very well together. You can almost consider them friends.